src/pip/_internal/index.py

```diff
@@ -43,7 +43,7 @@
     from pip._internal.pep425tags import Pep425Tag
     from pip._internal.utils.hashes import Hashes
 
-    BuildTag = Tuple[Any, ...]  # either empty tuple or Tuple[int, str]
+    BuildTag = Union[Tuple[()], Tuple[int, str]]
```

Comment:
Unrelated change?

Reply:
This was to fix an older type annotation, which used `tuple()` and was changed to `()` as caught via `check-builtin-literals`.

This even made it possible to have a stricter annotation (as changed here).

Reply:
Cool. I saw some later changes that were obviously fixes to issues raised by the new checks, but hadn't made the link with these ones.